dungeon/src/contrib/hud/elements/GUICombination.java

```diff
@@ -103,6 +103,15 @@ public void drawDebug(final ShapeRenderer shapes) {
     this.combinableGuis.forEach(CombinableGUI::drawDebug);
   }
 
+  /**
+   * Returns the list of CombinableGUI elements that are part of this GUICombination.
+   *
+   * @return An ArrayList of CombinableGUI elements.
+   */
+  public ArrayList<CombinableGUI> combinableGuis() {
+    return this.combinableGuis;
+  }
```

Comment:
So etwas finde ich aus dem Blickwinkel von Softwarearchitektur fragwürdig: Hier wird ein internes Attribut direkt nach draußen gereicht und könnte problemlos außerhalb der Klasse verändert werden. 

Wenn ich sowieso auf die internen Elemente einer Klasse zugreifen will, warum mache ich mir dann überhaupt die Mühe, Daten zu kapseln? Oder warum mache ich die Attribute nicht einfach `public`?

Du hast vermutlich gute Gründe für diese neue Methode. Mittelfristig frage ich mich aber, ob hier nicht die Modellierung bzgl. der GUI-Sachen angepasst werden sollte.

Reply:
Hier bin ich mir nicht ganz sicher. Ich verstehe deinen Punkt. Aber wenn man schon solche Gruppierungen von Elementen hat (hier `GuiCombination`), braucht man trotzdem irgendwie den Zugang zu den einzelden Elementen. Weil es ist intersannt welche GUI grade offen ist. Ist der Spieler grade am Craften, an einer Kiste oder doch nur im Inventar. Ich kann aber sonst einfach Methoden hinzufügen wie:
```java
public Optional<CraftingGUI> getCraftingGUI();

public Optional<InventoryGUI> getOtherInventory(); // Wenn der Spieler ein anderes Inventar offen hat, als nur sein eigenes offen hat.
```
Aber dann würde es ein Problem geben wenn der Spieler mehr als 1 anderes Inventar offen hat.

Insgesamt bin ich kein Freund von dieser Klassen Architektur. Ich finde `CombinableGUI` sollte eher ein Interface werden, statt wie jetzt eine Abstrakte Oberklasse von `CraftingGUI` & `InventoryGUI`. Dann könnte man schön ein Interface erstellen für die `transferItem` oder `addItem` Methoden. Aber vlt hat das ja auch einen Grund das es so gemacht wurde?

Reply:
Alternative kann ich aber auch einfach eine Kopie von der Liste returnen.

Reply:
> Aber wenn man schon solche Gruppierungen von Elementen hat (hier GuiCombination), braucht man trotzdem irgendwie den Zugang zu den einzelnen Elementen. Weil es interessant sein kann welche GUI grade offen sind. 

ja, und das war vermutlich auch der hintergrund für diese struktur. 

was mich stört, ist dass das alles einfach wieder rausgereicht wird und die nutzer damit irgendwas anstellen können (und müssen). besser wäre imho, wenn die klasse methoden zur iteration oder abfrage o.ä. (was man halt so mit den elementen machen will/muss) anbieten würde.

> Alternative kann ich aber auch einfach eine Kopie von der Liste returnen.

lass mal, das löst das problem ja nicht wirklich. und du hast das ja auch nicht so modelliert und versuchst nur, das irgendwie zu nutzen ...

> Insgesamt bin ich kein Freund von dieser Klassen Architektur. Ich finde CombinableGUI sollte eher ein Interface werden, statt wie jetzt eine Abstrakte Oberklasse von CraftingGUI & InventoryGUI. Dann könnte man schön ein Interface erstellen für die transferItem oder addItem Methoden. Aber vlt hat das ja auch einen Grund das es so gemacht wurde?

Da sprichst Du was an, ja. Ich halte das mal in einem neuen Ticket fest.